src/SWP/Bundle/TemplateEngineBundle/Controller/MenuController.php

```diff
@@ -0,0 +1,228 @@
+
+            return $this->handleView(View::create($menu, 201));
+        }
+
+        return $this->handleView(View::create($form, 400));
+    }
+
+    /**
+     * Delete single menu.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Delete single menu",
+     *     statusCodes={
+     *         204="Returned on success.",
+     *         404="Menu not found",
+     *         422="Menu id is not number"
+     *     }
+     * )
+     * @Route("/api/{version}/templates/menus/{id}", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_templates_delete_menu")
+     * @Method("DELETE")
+     */
+    public function deleteAction($id)
+    {
+        $menu = $this->getMenu($id);
+        if (!$menu) {
+            throw new NotFoundHttpException('Menu with this id was not found.');
+        }
+
+        /** @var DocumentManager $dm */
+        $dm = $this->get('document_manager');
+        $dm->remove($menu);
+        $dm->flush();
+
+        return $this->handleView(View::create(null, 204));
+    }
+
+    /**
+     * Update single menu.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Update single menu",
+     *     statusCodes={
+     *         201="Returned on success.",
+     *         404="Menu not found",
+     *         422="Menu id is not number",
+     *         405="Method Not Allowed"
+     *     },
+     *     input="SWP\Bundle\TemplateEngineBundle\Form\Type\MenuType"
+     * )
+     * @Route("/api/{version}/templates/menus/{id}", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_templates_update_menu")
+     * @Method("PATCH")
+     */
+    public function updateAction(Request $request, $id)
+    {
+        $menu = $this->getMenu($id);
+        if (!$menu) {
+            throw new NotFoundHttpException('Menu with this id was not found.');
+        }
+
+        $form = $this->createForm(new MenuType(), $menu, [
+            'method' => $request->getMethod(),
+        ]);
+
+        $form->handleRequest($request);
+        if ($form->isValid()) {
+            /** @var DocumentManager $dm */
+            $dm = $this->get('document_manager');
+            $dm->flush();
+
+            $menu = $this->getMenu($id);
+
+            return $this->handleView(View::create($menu, 201));
+        }
+
+        return $this->handleView(View::create($form, 200));
+    }
+
+    /**
+     * @return null|object
+     */
+    private function getMenuRootNode()
+    {
+        $mp = $this->get('swp_template_engine.menu_provider');
+        $menuParent = $mp->getMenuParent();
+
+        return $menuParent;
+    }
+
+    /**
+     * @param $id
+     *
+     * @return Menu
+     *
+     * @throws UnprocessableEntityHttpException
+     */
+    private function getMenu($id)
+    {
+        if (!$id) {
+            throw new UnprocessableEntityHttpException('You need to provide menu Id (name).');
+        }
+
+        $mp = $this->get('swp_template_engine.menu_provider');
```

Comment:
just call it `$menuProvider`. Variable names should be descriptive. Exceptions are only those really common like `$em, $dm`.